geps/gep-3798/index.md

```diff
@@ -0,0 +1,130 @@
+# GEP-3798: Client IP-Based Session Persistence
+
+* Issue: [#3798](https://github.com/kubernetes-sigs/gateway-api/issues/3798)
+* Status: Provisional
+
+(See [status definitions](../overview.md#gep-states).)
+
+## TLDR
+
+### What
+ This GEP proposes the addition of Client IP-based session persistence to the Gateway API. This feature will allow Gateway API implementations to ensure that requests originating from a specific client IP address (or a subnet defined by an IP mask) are consistently routed to the same backend endpoint for a configurable duration. This aims to provide a standardized and centralized mechanism for client IP persistence across various Gateway API implementations.
+
+## Goals
+
+* Define an API extension within Gateway API to enable client IP-based session persistence.
+
+* Allow configuration of a session duration for which a client IP should stick to a backend.
+
+* Provide an optional mechanism to specify an IP mask for subnet-based persistence, allowing multiple clients from the same subnet to be routed to the same backend.
```

Comment:
How broadly is this implementable? Generally for new Gateway API features, we're looking for at least 3 implementations, and strongly prefer supportability by multiple underlying dataplanes.

Reply:
I have added some examples where subnet mask is configurable in References section.